src/Filters/stream_filt.jl

```diff
@@ -163,12 +159,12 @@ function FIRFilter(h::Vector, resampleRatio::Union{Integer,Rational} = 1)
     if resampleRatio == 1                                     # single-rate
         kernel     = FIRStandard(h)
         historyLen = kernel.hLen - 1
-    elseif interpolation == 1                                 # decimate
-        kernel     = FIRDecimator(h, decimation)
-        historyLen = kernel.hLen - 1
     elseif decimation == 1                                    # interpolate
         kernel     = FIRInterpolator(h, interpolation)
         historyLen = kernel.tapsPerϕ - 1
+    elseif interpolation == 1                                 # decimate
+        kernel     = FIRDecimator(h, decimation)
+        historyLen = kernel.hLen - 1
```

Comment:
Why this change?

Reply:
Aha, yes. For integer ratio, `resampleRatio != 1` implies `interpolation != 1` (because they are equal), so we cannot get an `FIRDecimator`, but inference doesn't do this kind of analysis. OTOH, `decimation` is inferred as constant 1, so with the reordered tests, inference can deduce that we'll get an `FIRInterpolator` if we have already gotten an `FIRStandard`. Alright, nice little improvement.